test_conformance/basic/test_vector_creation.cpp

```diff
@@ -171,236 +187,319 @@ int create_kernel(ExplicitType type, int output_size, char *program, int *number
+        // the right. We do this by causing those loops to skip on the next
+        // iteration.
+        if (vloads < DEPTH)
+        {
             if (DEBUG > 1) log_info("done with this depth\n");
-            for (int k=vloads; k<DEPTH; k++)
-                pos[k] = number_of_sizes;
+            for (int k = vloads; k < DEPTH; k++) pos[k] = number_of_sizes;
         }
 
         // Increment the far right size by 1, rolling over as needed
-        for (int d=DEPTH-1; d>=0; d--) {
+        for (int d = DEPTH - 1; d >= 0; d--)
+        {
             pos[d]++;
-            if (pos[d] >= number_of_sizes) {
+            if (pos[d] >= number_of_sizes)
+            {
                 pos[d] = 0;
-                if (d == 0) {
+                if (d == 0)
+                {
                     // If we rolled over at the far-left then we are done
                     done = 1;
                     break;
                 }
-            } else {
+            }
+            else
+            {
                 break;
             }
         }
-        if (done)
-            break;
+        if (done) break;
 
         // Continue until we are done.
     }
-    strcat(program, "}\n\n"); //log_info("%s\n", program);
+    strcat(program, "}\n\n"); // log_info("%s\n", program);
     total_program_length += 3;
-    if (DEBUG) log_info("\t\t(Program for vector type %s%s contains %d vector creations, of total program length %gkB, with a total of %d vloads.)\n",
-                        get_explicit_type_name(type), size_names[number_of_sizes-1], total_results, total_program_length/1024.0, total_vloads);
+    if (DEBUG)
+        log_info(
+            "\t\t(Program for vector type %s%s contains %d vector creations, "
+            "of total program length %gkB, with a total of %d vloads.)\n",
+            get_explicit_type_name(type), size_names[number_of_sizes - 1],
+            total_results, total_program_length / 1024.0, total_vloads);
     *number_of_results = current_result;
-    if (aborted_due_to_size)
-        return total_results;
+    if (aborted_due_to_size) return total_results;
     return 0;
 }
 
 
-
-
-int test_vector_creation(cl_device_id deviceID, cl_context context, cl_command_queue queue, int num_elements)
+int test_vector_creation(cl_device_id deviceID, cl_context context,
+                         cl_command_queue queue, int num_elements)
 {
-    ExplicitType vecType[] = { kChar, kUChar, kShort, kUShort, kInt, kUInt, kLong, kULong, kFloat, kDouble };
-    unsigned int vecSizes[] = { 1, 2, 3, 4, 8, 16};
+    const std::vector<ExplicitType> vecType = { kChar,  kUChar, kShort, kUShort,
+                                                kInt,   kUInt,  kLong,  kULong,
+                                                kFloat, kHalf,  kDouble };
+    // should be in sync with global array size_names
+    const std::vector<unsigned int> vecSizes = { 1, 2, 3, 4, 8, 16 };
```

Comment:
This is so minor it probably isn't worth changing, but I was initially confused because this test now outputs e.g.

```sh
Testing half.
Testing half2...
Testing half3...
Testing half4...
Testing half8...
```

Then, since this array of `vecSizes` includes `1`, so it's easy to incorrectly assume that scalars are also tested.  The loop that iterates over this array properly starts at index `1` though, so the `vecSize == 1` case is properly skipped, and `Testing half.` is only generic output that "half" vectors are being tested.

I was originally thinking that the value `1` could simply be removed from this array, but the `size_index` is also used to index into some other arrays, so additional changes will be required.  Probably not worth it.

The one change I _might_ consider is to put a bogus value here instead, so it is more clear that it is unused, but even that might be more trouble than it's worth.